cypress/integration/examples/code-highlighting.ts

```diff
@@ -0,0 +1,35 @@
+describe('code highlighting', () => {
+  const slateEditor = '[data-slate-node="element"]'
+  const leafNode = 'span[data-slate-leaf="true"]'
+
+  beforeEach(() => {
+    cy.visit('examples/code-highlighting')
+  })
+
+  it('highlights HTML tags', () => {
+    cy.get(slateEditor)
+      .find('span')
+      .eq(0)
+      .find(leafNode)
+      .eq(0)
+      .should('contain', '<h1>')
+      .should('have.css', 'color', 'rgb(153, 0, 85)')
+  })
+
+  it('highlights javascript syntax', () => {
+    const JSCode = 'const slateVar = 30;{enter}'
+    cy.get('select').select('JavaScript') // Select the 'JavaScript' option
+
+    cy.get(slateEditor)
+      .type('{movetostart}')
+      .type(JSCode) // Type JavaScript code
+
+    cy.get(slateEditor)
+      .find('span')
+      .eq(0)
+      .find(leafNode)
+      .eq(0)
+      .should('contain', 'const')
```

Comment:
This test is causing several new PRs to fail, specifically #3888 and #4037:

```bash
  code highlighting
    ✓ highlights HTML tags (16743ms)
event - build page: /examples/[example]
wait  - compiling...
    1) highlights javascript syntax


  1 passing (39s)
  1 failing

  1) code highlighting
       highlights javascript syntax:
     AssertionError: Timed out retrying after 4000ms: expected '<span.css-14qid8c>' to contain 'const'
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/examples/code-highlighting.ts:124:14)
```

Is it new code that's broken or is does this test not appear to be accurate?

Reply:
I think we should increase the timout, the test is new one but is fine. Might be unable to render the changes within the current timeframe.